### PR TITLE
Develop layout_stop-watch

### PR DESCRIPTION
**Pull request by sergey-lipikhin**
- [DEMO LINK](https://sergey-lipikhin.github.io/layout_stop-watch/)

### CODE REVIEW

File: src/styles/blocks/watch.scss [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+$big-hand-width: size-with-scale(10px);
+$big-hand-height: size-with-scale(300px);
+
+@mixin hand-animation($duration) {
+  animation: tic $duration linear;
+  animation-iteration-count: infinite;
+  transform-origin: bottom center;
+}
+
+%hand-position {
```
Comment (Andr1yK) [inline src/styles/blocks/watch.scss:22]: this is a mess

Comment (Andr1yK) [reply src/styles/blocks/watch.scss:22]: variables, functions, styles in one place

**Review by Andr1yK — CHANGES_REQUESTED** [review]

File: src/styles/blocks/watch.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    width: $center-radius * 2;
+    height: $center-radius * 2;
+    border-radius: $center-radius;
+    background-color: palevioletred;
+  }
+
+  &__small-hand {
+    @extend %hand-position;
+    @include hand-animation(60s);
```
Comment (sergii-nosachenko) [inline src/styles/blocks/watch.scss:13]: Wrong timing for minute hand

File: src/styles/blocks/watch.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    width: $small-hand-width;
+    height: $small-hand-height;
+    background-color: blue;
+  }
+
+  &__long-hand {
+    @extend %hand-position;
+    @include hand-animation(30s);
```
Comment (sergii-nosachenko) [inline src/styles/blocks/watch.scss:22]: Wrong timing for seconds hand

File: src/styles/utils/_animations.scss [hunk 4]
```diff
@@ -0,0 +1,8 @@
+@keyframes tic {
+  from {
+    transform: translateX(-50%) rotate(0deg);
```
Comment (sergii-nosachenko) [inline src/styles/utils/_animations.scss:3]: Why do you need `translateX(-50%)` if it doesn't change?

File: src/styles/blocks/watch.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
+.watch {
+  &__center {
+    position: relative;
+
+    width: $center-radius * 2;
+    height: $center-radius * 2;
+    border-radius: $center-radius;
+    background-color: palevioletred;
```
Comment (sergii-nosachenko) [inline src/styles/blocks/watch.scss:8]: Add all colors to vars

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/styles/utils/_animations.scss (hunk 4 above)
Comment (sergey-lipikhin) [reply src/styles/utils/_animations.scss]: Initially I use `top: 50%; left: 50%; transform: translateX(-50%);` to center hands over the watch.

But when the animation starts it rewrites `transform` to `transform: rotate(0deg);` and the x-offset becomes default so hands are not centralized.

To maintain hands center position along animation I add `translateX(-50%)` to every iteration. And it looks like this:
`transform: translateX(-50%) rotate(0deg);`

File: src/styles/utils/_animations.scss [hunk 6]
```diff
@@ -0,0 +1,8 @@
+@keyframes tic {
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
```
Comment (sergii-nosachenko) [inline src/styles/utils/_animations.scss:7]:
```suggestion
  to {
    transform: rotate(1turn);
  }
```

**Review by sergii-nosachenko — APPROVED** [review]
Good work 👍

**Pull request closed without merging**